src/singletons.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { PathType, PropertiesType } from "./types.js";
+
+export let Path: PathType = (null: any);
+export let Properties: PropertiesType = (null: any);
+
+export function setPath(singleton: PathType) {
+  Path = singleton;
```

Comment:
Maybe add an invariant like `invariant(singleton === Path || singleton === undefined)`, as these functions might get called multiple times, somewhat surprisingly.

Reply:
I'll add the invariants in the in a later pull request.

Reply:
It turns out that enforcing such an invariant is way more trouble than it is worth. The gain is at best extremely modest. The aggravation of maintaining the invariant is not.